.htaccess

```diff
@@ -24,3 +24,11 @@ Redirect permanent /datafusion-python https://datafusion.apache.org/python
 
 # redirect all ballista URLs to new website
 Redirect permanent /ballista https://datafusion.apache.org/ballista
+
+# enable kapa.ai bot (GH-45665)
+# See https://docs.kapa.ai/integrations/understanding-csp-cors
+<IfModule mod_headers.c>
```

Comment:
I think that we don't need `<IfModule mod_headers.c>` because `mod_headers` is always available on apache.org.

BTW, this `.htaccess` may not work... It seems that @ianmcook already tried this approach: https://github.com/apache/arrow-site/pull/599

Reply:
> I think that we don't need <IfModule mod_headers.c> because mod_headers is always available on apache.org.

In practice I bet you're right but I've always preferred to guard module-dependent directives out of caution.

And yeah, in the previous PRs, it wasn't clear to me how ASF was ignoring our custom CSP but I think in Apache you can disallow certain directives in .htaccess files but, if they're doing something else, this may work.